.gitignore

```diff
@@ -1 +1,29 @@
-mqtt-broker
+# Allowlisting gitignore template for GO projects prevents us
+# from adding various unwanted local files, such as generated
+# files, developer configurations or IDE-specific files etc.
+#
+# Recommended: Go.AllowList.gitignore
+
+# Ignore everything
+*
+
+# But not these files...
+!/.gitignore
+
+
+!*.go
+!go.sum
+!go.mod
+
+!/README.md
+!/CHANGELOG.md
+!/LICENSE
+
+!/.golangci.yaml
+# !/.pre-commit-config.yaml
+!/Makefile
```

Comment:
nit: why we need this? We are gonna build our project using make? 
Since I assume it to be a small project, I would thing go build would be enough.

Reply:
`Makefile` has all the commands to build, test, run checks etc.

Just a single location to call from when we integrate to GitHub workflows.

Just `make <command>` in the github workflows YAML conf.